committeeoversightapp/views.py

```diff
@@ -1,3 +1,160 @@
+from opencivicdata.core.models import Organization
+from .forms import EventForm, CommitteeForm, CommitteeMemberForm, WitnessForm, TranscriptForm
+
+class Success(TemplateView):
+    template_name = 'success.html'
+
+class EventView(DetailView):
+    model = Event
+    template_name = 'detail.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        return context
+
+# given a url string, find the file extension at the end
+def get_ext(url):
+    path = urlparse(url).path
+    ext = splitext(path)[1]
+    return ext
+
+# archive a url string
+def archive_url(url):
+    wayback_host = 'http://web.archive.org'
+    save_url = '{0}/save/{1}'.format(wayback_host, url)
+    archived = requests.get(save_url)
+    archive_url = '{0}{1}'.format(wayback_host, archived.headers['Content-Location'])
+    return archive_url
+
+class EventCreate(TemplateView):
+    template_name = "create.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['event_form'] = EventForm(prefix="event")
+        context['committee_form'] = CommitteeForm(prefix="committee")
+        context['committeemember_form'] = CommitteeMemberForm(prefix="committeemember")
+        context['witness_form'] = WitnessForm(prefix="witness")
+        context['transcript_form'] = TranscriptForm(prefix="transcript")
+
+        return context
+
+    def post(self, request, **kwargs):
+        event_form = EventForm(request.POST, prefix="event")
+        committee_form = CommitteeForm(request.POST, prefix="committee")
+        committeemember_form = CommitteeMemberForm(request.POST, prefix="committeemember")
+        witness_form = WitnessForm(request.POST, prefix="witness")
+        transcript_form = TranscriptForm(request.POST, prefix="transcript")
+
+        print("Checking if forms are valid...")
+
+        forms_valid = [event_form.is_valid(), committee_form.is_valid(),
+                       committeemember_form.is_valid(), witness_form.is_valid(),
+                       transcript_form.is_valid()]
+
+        if all(forms_valid):
+            print("All forms valid! Saving...")
+
+            # save event
+            event = event_form.save()
+
+            # find and create committees as EventParticipants
+            committees = committee_form.cleaned_data['name']
+
+            for committee in committees:
+                name = committee.name
+                event = event_form.save(commit=False)
+                organization = Organization.objects.get(id=committee.id)
+                entity_type = "organization"
+                new_committee = EventParticipant(name=name, event=event, organization=organization, entity_type=entity_type)
+                new_committee.save()
+
+            # find and create committee members as EventParticipants
+            committeemembers = committeemember_form.cleaned_data['name'].split(",")
+
```

Comment:
I am not persuaded that this is the best way to do this, i.e., by asking the user to input a comma-separated list. (Is there a reason you've implemented it that way? did the folks at Lugar suggest something along those lines?)

This seems like a good opportunity for a [formset](https://docs.djangoproject.com/en/2.1/topics/forms/formsets/), which would allow the user to create several committee members and witnesses (helping you avoid the awkwardness of cleaning up after calling `split`). Happy to discuss this face-to-face!

Reply:
I'd started with a formset but it seemed clunky if the data enterer is started with a single string. I asked about formatting for this and the Witnesses field in my email to Jamie this morning, will adjust accordingly!

Reply:
Terrific. I think let's leave this for now, but can you open an issue about it? Then, we will hear from Jamie, we'll decide the best course of action.